### PR TITLE
Provide the --freeze parameter. Fixes #385

### LINKED ISSUE #385: Include `--freeze`
Example command: `fades -r requirements.txt --freeze=freezedreqs.txt -x pytest -v -s`

It should `(pip) freeze` the just created venv to the specified file. If the venv is used from the cache, and not re-created, the freeze is a noop.

### PR DESCRIPTION
**Pull request by facundobatista**

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -305,6 +305,52 @@ matter which one is the latest. But if you do::
...
+project, you actually get to use a new virtualenv automatically.
+
+If you don't pin the dependencies in your requirements file, this has 
+another nice side effect: everytime you use them in a new environment (or
+if you have `--check-updates` set) you will get latest versions, effectively
+avoiding the trap of sticking in old versions forever.
+
+However, this has a bad side. If it happens that a dependency of your 
+project released a revision between the moment you run the tests and the 
+mement your project is deployed to the server, it may happen that you
```
Comment (gilgamezh) [inline README.rst:321]: mement

File: fades/pipmanager.py [hunk 2]
```diff
@@ -108,3 +108,10 @@ def _brute_force_install_pip(self):
         python_exe = os.path.join(self.env_bin_path, "python")
         helpers.logged_exec([python_exe, self.pip_installer_fname, '-I'])
         self.pip_installed = True
+
+    def freeze(self, filepath):
+        """Dump venv contents to the indicated filepath."""
+        logger.debug("running freeze to store in %r", filepath)
+        stdout = helpers.logged_exec([self.pip_exe, "freeze", "--all", "--local"])
+        with open(filepath, "wt", encoding='utf8') as fh:
```
Comment (gilgamezh) [inline fades/pipmanager.py:116]: why are you adding `t` here? That's just the default

File: README.rst (hunk 1 above)
Comment (facundobatista) [reply README.rst]: Gotcha

File: fades/pipmanager.py (hunk 2 above)
Comment (facundobatista) [reply fades/pipmanager.py:116]: Explicit better than implicit :).

Now seriously, I always write 't' or 'b' there, as the difference is getting bytes or unicode chars, which may be critical in the following handling of the file handler.

**Pull request merged**